cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityWritePrivilegesTest.scala

```diff
@@ -14,23 +14,208 @@ class SecurityWritePrivilegesTest extends DocumentingTest with QueryStatisticsTe
+      }
+
+      p("The `REMOVE LABEL` privilege can also be denied.")
+      p("include::deny-remove-label-syntax.asciidoc[]")
+
+      p("For example, denying the ability to remove the label `foo` from nodes of all graphs to the role `regularUsers` would be achieved using:")
+      query("DENY REMOVE LABEL foo ON GRAPH * TO regularUsers", ResultAssertions(r => {
+        assertStats(r, systemUpdates = 1)
+      })) {
+        statsOnlyResultTable()
+      }
+    }
+
+    section("The `SET PROPERTY` privilege", "administration-security-writes-set-property", "enterprise-edition") {
+      p(
+        """The `SET PROPERTY` privilege enables a user to set a property on a node or relationship element in a graph using the <<set-set-a-property, SET clause>>.""".stripMargin)
+      p("include::grant-set-property-syntax.asciidoc[]")
+
+      p(
+        """For example, granting the ability to set any property on all elements of the graph `neo4j` to the role `regularUsers` would be achieved using:""".stripMargin)
+      query("GRANT SET PROPERTY {*} ON GRAPH neo4j ELEMENTS * TO regularUsers", ResultAssertions(r => {
+        assertStats(r, systemUpdates = 2)
+      })) {
+        statsOnlyResultTable()
+      }
+
+      p("The `SET PROPERTY` privilege can also be denied.")
+      p("include::deny-set-property-syntax.asciidoc[]")
+
+      p("For example, denying the ability to set the property `foo` on nodes with the label `bar` on all graphs to the role `regularUsers` would be achieved using:")
+      query("DENY SET PROPERTY { foo } ON GRAPH * NODES bar TO regularUsers", ResultAssertions(r => {
+        assertStats(r, systemUpdates = 1)
+      })) {
+        statsOnlyResultTable()
+      }
+      note {
+        p("If the users attempts to set a property with a property name that does not already exist on the database the user must also possess the " +
+          "<<administration-security-administration-database-tokens, CREATE NEW PROPERTY NAME>> privilege.")
+      }
+    }
+
+    section("The `MERGE` privilege", "administration-security-writes-merge", "enterprise-edition") {
+      p(
+        """The `MERGE` privilege is a compound privilege that combines `TRAVERSE` and `READ` (i.e. `MATCH`) with `CREATE` and `SET PROPERTY`. This is intended to
+          | permit use of <<query-merge, the MERGE command>> but is applicable to all reads and writes that require these privileges.""".stripMargin)
+      p("include::grant-merge-syntax.asciidoc[]")
+
+      p(
+        """For example, granting `MERGE` on all elements of the graph `neo4j` to the role `regularUsers` would be achieved using:""".stripMargin)
+      query("GRANT MERGE {*} ON GRAPH neo4j ELEMENTS * TO regularUsers", ResultAssertions(r => {
+        assertStats(r, systemUpdates = 2)
+      })) {
+        statsOnlyResultTable()
+      }
+
+      p("It is not possible to deny the `MERGE` privilege. If it is desirable to prevent a users from creating elements and setting properties use " +
```

Comment:
```suggestion
      p("It is not possible to deny the `MERGE` privilege. If it is desirable to prevent a users from creating elements and setting properties, use " +
```